### PR TITLE
Additional enumerations for SFA/MF

### PR DESCRIPTION
**Pull request by shorowit**
Adds a few AdjacentTo/Location enumerations useful for single-family attached and multifamily housing. For ERI, this allows mapping as follows:
- Multifamily Buffer Boundary => "other multifamily buffer space" (new, e.g. enclosed unconditioned stairwell)
- Non-Freezing Space => "other non-freezing space" (new, e.g. hared parking garage ceiling)
- Unrated Heated Space => "other heated space" (new, e.g. shared laundry/equipment space)
- Unrated Conditioned Space => "other housing unit" (existing, living/conditioned space of adjacent unit)

### CODE REVIEW

File: schemas/HPXMLDataTypes.xsd [hunk 1]
```diff
@@ -2172,6 +2181,12 @@
 			<xs:enumeration value="other exterior"/>
 			<xs:enumeration value="roof deck"/>
 			<xs:enumeration value="unconditioned space"/>
+			<xs:enumeration value="other heated space"/>
```
Comment (yzhou601) [inline schemas/HPXMLDataTypes.xsd]: UnitLocation type is used in `WaterHeatingSystem/Location`, `HVAC/HVACPlant/HeatingSystem | CoolingSystem | HeatPump/UnitLocation`, and `Project/ProjectDetails/Measures/Measure/Location`. I think for these usage, it's probably ok to directly add new enumerations. Does it make sense?

**Comment by yzhou601** [comment]
> @yzhou601 Can you preserve the alphabetical order in these lists? Also, there's no need to include "other housing unit above" and "other housing unit below" in these lists (those are available in the other enumeration because they are needed for the FrameFloor AdjacentTo choices).

Sure, will do.

File: schemas/HPXMLDataTypes.xsd (hunk 1 above)
Comment (shorowit) [reply schemas/HPXMLDataTypes.xsd]: That seems reasonable.

**Comment by shorowit** [comment]
Now:
- [ ] Document that "other housing unit" refers to the living/conditioned space of the adjacent unit.

Future:
- [ ] Consider a new enumeration like "other housing unit, same space" (or something better)
- [ ] Consider renaming "other housing unit" to be clearer per the description above

**Comment by nmerket** [comment]
:movie_camera: [Discussion in HPXML Working Group Meeting 2023-07-11](https://youtu.be/0vfe3srBLts?t=552)

**Pull request merged**